### DISCUSSION TITLE
Carousel breaks when API data ingested

### DISCUSSION THREAD

**Discussion by ilmoi** [davidjerleke/embla-carousel#444] [Help] [answered]
Running into a weird problem with embla-carousel-react.

Whenever the carousel depends on data pulled in via the API, it non-deterministically breaks. Specifically, the carousel has the correct number of items (>1), but scrolling doesn't work (neither with buttons, nor by dragging). 

If the data is hardcoded beforehand this doesn't happen. So it seems something breaks because it goes from 0 items to eg 10.

Has anyone run into this before? Any advice? Would hate to move to another package, but this got so annoying we either need to solve it or we dont have a choice.

Our versions:

"embla-carousel-react": "^7.0.9",
"embla-carousel-autoplay": "^7.1.0",

**Comment by davidjerleke** [comment, marked as answer]
Hi @ilmoi,

From the Embla [documentation](https://www.embla-carousel.com/guides/add-and-remove-slides/#with-react):

```ts
useEffect(() => {
  if (emblaApi) emblaApi.reInit()
}, [emblaApi, slides])
```

Best,
David

**Reply by davidjerleke** [reply]
@ilmoi, did you manage to solve the problem with my [help](https://github.com/davidjerleke/embla-carousel/discussions/444#discussioncomment-5325837)?

**Reply by ilmoi** [reply]
it worked! Thanks @davidjerleke. One piece of feedback: I would make that way more prominenet in the docs, ideall part of "get started". I spent ages on the docs and didn't think to click into this page, and it was super frustrating why such basic functionality doesn't work. Surely the default assumption for most users is their slide numbers can change. Either way - thanks for the help.

**Reply by davidjerleke** [reply]
- #159.

The feature is on its [way](https://github.com/davidjerleke/embla-carousel/issues/159#issuecomment-1474765687).

**Reply by davidjerleke** [reply]
Hi @ilmoi, 

This feature has been released with [v8.0.0-rc01](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.0.0-rc01).

Best,
David
